optashift-employee-rostering-gwtui/src/main/java/org/optaplanner/openshift/employeerostering/gwtui/client/pages/employeeroster/EmployeeBlob.java

```diff
@@ -16,143 +16,103 @@
 
-public class EmployeeBlob implements Blob<OffsetDateTime> {
+public class EmployeeBlob extends TimeslotBlob {
 
-    private final LinearScale<OffsetDateTime> scale;
-    private EmployeeAvailability employeeAvailability;
-    private Shift shift;
-    private Long sizeInGridPixels;
+    private EmployeeAvailabilityView employeeAvailabilityView;
+    private ShiftView shiftView;
+    private Map<Long, Spot> spotIdToSpotMap;
+    private Map<Long, Employee> employeeIdToEmployeeMap;
 
-    private Long positionInGridPixelsCache = null;
-    private Long endPositionInGridPixelsCache = null;
-
-    public EmployeeBlob(final LinearScale<OffsetDateTime> scale, final Shift shift) {
-        this(scale, null, shift);
+    public EmployeeBlob(final LinearScale<LocalDateTime> scale, Map<Long, Spot> spotIdToSpotMap,
+                        Map<Long, Employee> employeeIdToEmployeeMap, final ShiftView shiftView) {
+        this(scale, spotIdToSpotMap, employeeIdToEmployeeMap, null, shiftView);
     }
 
-    public EmployeeBlob(final LinearScale<OffsetDateTime> scale, final EmployeeAvailability employeeAvailability) {
-        this(scale, employeeAvailability, null);
+    public EmployeeBlob(final LinearScale<LocalDateTime> scale, Map<Long, Spot> spotIdToSpotMap,
+                        Map<Long, Employee> employeeIdToEmployeeMap, final EmployeeAvailabilityView employeeAvailabilityView) {
+        this(scale, spotIdToSpotMap, employeeIdToEmployeeMap, employeeAvailabilityView, null);
     }
 
-    private EmployeeBlob(final LinearScale<OffsetDateTime> scale, final EmployeeAvailability employeeAvailability, final Shift shift) {
-        this.employeeAvailability = employeeAvailability;
-        this.shift = shift;
-        this.scale = scale;
-        this.sizeInGridPixels = scale.toGridPixels(getEndDateTime()) - scale.toGridPixels(getStartDateTime());
+    private EmployeeBlob(final LinearScale<LocalDateTime> scale, Map<Long, Spot> spotIdToSpotMap,
+                         Map<Long, Employee> employeeIdToEmployeeMap, final EmployeeAvailabilityView employeeAvailabilityView, final ShiftView shiftView) {
+        this.employeeAvailabilityView = employeeAvailabilityView;
+        this.shiftView = shiftView;
+        this.spotIdToSpotMap = spotIdToSpotMap;
+        this.employeeIdToEmployeeMap = employeeIdToEmployeeMap;
+        setScale(scale);
     }
 
-    private OffsetDateTime getStartDateTime() {
-        if (null != shift) {
-            return shift.getStartDateTime();
+    public String getLabel() {
+        if (shiftView != null) {
+            return getSpot().toString() + ": " + shiftView.getStartDateTime() + "-" + shiftView.getEndDateTime();
         } else {
-            return employeeAvailability.getStartDateTime();
+            return employeeAvailabilityView.getState().toString() + ":" + employeeAvailabilityView.getStartDateTime() + "-" + employeeAvailabilityView.getEndDateTime();
```

Comment:
on enum's, use name() instead of toString() if it's for the UI

Reply:
Javadoc says "Most programmers should use the toString() method in preference to this one, as the toString method may return a more user-friendly name" for the `name()` method, are you sure?